src/io.cr

```diff
@@ -1146,7 +1146,7 @@ abstract class IO
   end
 
   # Same as `pos`.
-  def tell
+  def tell : Int64 | Int32
```

Comment:
This look very odd to me. Especially as pos is noreturn. Where does this union type come from?

Reply:
I see [Memory#pos](https://github.com/crystal-lang/crystal/blob/master/src/io/memory.cr#L320) return Int32 and [Buffered#pos](https://github.com/crystal-lang/crystal/blob/master/src/io/buffered.cr#L203) return Int64. The side effect being this wrapping def indirectly inherits the return types of those methods :shrug: 

Maybe better to omit the return type restriction in this case?

Reply:
Either that, or declare it to be typeof(pos), if that is possible 🤔